File: 1122Programming/leetcode/1. Two Sum/ans.cpp
/*
 *
 * https://leetcode.com/problems/two-sum/
 * Problem ID: 1. Two Sum
 * 
 * Author: Ateto
 * 
 */

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int sz = nums.size();
        for(int i=0; i<sz; i++) {
            for(int j=i+1; j<sz; j++) {
                if(nums[i] + nums[j] == target) return {i, j};
            }
        }
        return {-1, -1};
    }
};

File: 1122Programming/leetcode/1. Two Sum/ans.rs
/*
 *
 * https://leetcode.com/problems/two-sum/
 * Problem ID: 1. Two Sum
 * 
 * Author: revival0728
 * 
 */

use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut map: HashMap<i32, usize> = HashMap::new();
        for (id, v) in nums.iter().enumerate() {
            if(map.contains_key(&(target - v))) {
                return vec![id.try_into().unwrap(), (*map.get(&(target - v)).unwrap()).try_into().unwrap()];
            } else {
                map.insert(*v, id);
            }
        }
        return vec![-1, -1];
    }
}
